### PR TITLE
refactor(heap_profiling): use background heap profiling and enable auto dump for compute and compactor

### PR DESCRIPTION
**Pull request by yuhao-su**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- use background heap profiling and enable auto dump for compute and compactor

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]
LGTM!

**Comment by fuyufjh** [comment]
Oh. You might need 

https://github.com/risingwavelabs/risingwave/blob/fbd54d89b05a27f9f8da803e0e29d3b0c4b2e366/src/compute/src/lib.rs#L125-L128

for compactor node as well

File: src/compute/src/server.rs [hunk 1]
```diff
@@ -284,19 +285,19 @@ pub async fn compute_node_serve(
...
     tokio::spawn(memory_mgr.clone().run(
         batch_mgr_clone,
         stream_mgr_clone,
         system_params.barrier_interval_ms(),
         system_params_manager.watch_params(),
     ));
 
+    let heap_profiler = HeapProfiler::new(total_memory_bytes, config.server.heap_profiling.clone());
+    // Run a background heap profiler
+    tokio::spawn(heap_profiler.start());
```
Comment (BugenZhao) [inline src/compute/src/server.rs:299]: I'm not sure if dumping heap can be blocking and interfere other tasks running on the main runtime. If so, we may consider spawning a new thread.

**Review by BugenZhao — APPROVED** [review]

**Review by BugenZhao — COMMENTED** [review]
Is it possible that we start compactor and compute services simultaneously in the standalone mode? There'll be two background heap profiler running then.

File: src/compute/src/server.rs (hunk 1 above)
Comment (yuhao-su) [reply src/compute/src/server.rs:299]: I'm not sure, but it works well before refactor, which is also subject to the problem if it's blocking.

**Comment by yuhao-su** [comment]
> Is it possible that we start compactor and compute services simultaneously in the standalone mode? T

I guess in this case we should only have 1 profiler for each process.

**Comment by BugenZhao** [comment]
> > Is it possible that we start compactor and compute services simultaneously in the standalone mode? T
> 
> I guess in this case we should only have 1 profiler for each process.

Really? Looks like both `start` will be called.

**Comment by BugenZhao** [comment]
I think the most ideal way is to start the service in `main_okk`. However, if you find it tricky since it relies on reading the config file, I guess a quick hack is okay. FYI, this is how #11812 handles singleton service for standalone deployment. (cc @wenym1)

https://github.com/risingwavelabs/risingwave/blob/456f3e98d7a00e3c383f8b03ee40b5da09af58d9/src/common/common_service/src/metrics_manager.rs#L30-L61

**Pull request merged**